File: go/zero_matrix.go
package main

import (
  "fmt"
  "math/rand"
)


func valueInArray(array []int, value int) bool {
  if len(array) == 0 {
    return false
  }

  for _, num := range array {
    if num == value {
      return true
    }
  }
  return false
}


func zero(matrix [][]int) [][]int {
  rows := make([]int, 0)
  columns := make([]int, 0)

  for i := 0; i < len(matrix); i++ {
    for j := 0; j < len(matrix[i]); j++ {
      if matrix[i][j] == 0 {
        rows = append(rows, i)
        columns = append(columns, j)
      }
    }
  }

  for _, rowNum := range rows {
    row := matrix[rowNum]

    for j := range row {
      matrix[rowNum][j] = 0
    }
  }

  for _, columnNum := range columns {
    for i := range matrix {
      matrix[i][columnNum] = 0
    }
  }

  return matrix
}


func main() {
  var M int = 6
  var N int = 8
  rand.Seed(11)

  matrix := make([][]int, M)

  for i := 0; i < M; i++ {
    matrix[i] = make([]int, N)
    for j := 0; j < N; j++ {
      matrix[i][j] = rand.Intn(30)
    }
  }

  fmt.Println(matrix)
  fmt.Println(zero(matrix))
}

File: Arrays_Strings/python/zero_matrix.py
from random import randint


def zero_matrix(matrix):
    rows = []
    columns = []

    for i in range(len(matrix)):
        for j in range(len(matrix[i])):
            if matrix[i][j] == 0:
                rows.append(i)
                columns.append(j)

    for row_num in rows:
        row = matrix[row_num]

        for j in range(len(row)):
            matrix[row_num][j] = 0

    for column_num in columns:
        for i in range(len(matrix)):
            matrix[i][column_num] = 0


    return matrix


def main():
    m = 2
    n = 3

    matrix = [[0]*n for i in range(m)]

    for i in range(m):
        for j in range(n):
            matrix[i][j] = randint(0, 5)

    print(matrix)
    print(zero_matrix(matrix))

if __name__ == '__main__':
    main()
